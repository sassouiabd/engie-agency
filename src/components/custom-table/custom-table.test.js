import React from "react";
import { render, cleanup } from "@testing-library/react";
import CustomTable from "./custom-table.component";
import * as reactRedux from "react-redux";

describe("CustomTable component", () => {
  const useSelectorMock = jest.spyOn(reactRedux, "useSelector");
  const useDispatchMock = jest.spyOn(reactRedux, "useDispatch");
  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
  });

  afterEach(cleanup);

  it("match snapshot reference", () => {
    const dummyDispatch = jest.fn();
    useDispatchMock.mockReturnValue(dummyDispatch);

    expect(dummyDispatch).not.toHaveBeenCalled();

    const { asFragment } = render(<CustomTable />);

    expect(asFragment(<CustomTable />)).toMatchSnapshot();
  });
});
