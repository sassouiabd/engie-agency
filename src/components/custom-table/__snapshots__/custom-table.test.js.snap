// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTable component match snapshot reference 1`] = `<DocumentFragment />`;
